include/networkit/graph/GraphTools.hpp

```diff
@@ -418,6 +419,63 @@ void sortEdgesByWeight(Graph &G, bool decreasing = false);
  */
 std::vector<node> topologicalSort(const Graph &G);
 
+/**
+ * Randomizes the weights of the given graph. The weights are uniformly distributed in
+ * the range [0, 1] by default, unless a different distribution is provided. However it
+ * is only strictly in-place for already weighted graphs. For unweighted graphs a copy is
+ * created before randomizing weights.
+ *
+ * @param   G           Directed input graph
+ * @param   distr       Random distribution
+ */
+template <class Distribution = std::uniform_real_distribution<edgeweight>>
+void randomizeWeights(Graph &G, Distribution distr = std::uniform_real_distribution<edgeweight>{
+                                    0, std::nexttoward(1.0, 2.0)}) {
+    if (!G.isWeighted())
+        G = toWeighted(G);
+#pragma omp parallel
+    {
+        std::mt19937 gen;
+        auto baseSeed = Aux::Random::integer();
```

Comment:
Doesn't this make the function non-deterministic? I suggest to use `Aux::Random::getSeed()`.

Reply:
No, we est fr non-derminism with testEdgesRandomizerDeterminism

Reply:
Still I don't understand what is the reason to use a random number as the base seed while we have `Aux::Random::getSeed()`.

Reply:
We are starting one random number generator per thread. If all threads used the same seed we will have lots of duplicate weights.

Reply:
Good point. However, that's true only if (i) [`globalSeedGeneration`](https://github.com/networkit/networkit/blob/master/networkit/cpp/GlobalState.cpp#L9) is not 0 and if (ii) [`seedUseThreadId`](https://github.com/networkit/networkit/blob/master/networkit/cpp/GlobalState.cpp#L10) is false, see the [`getSeed` implementation](https://github.com/networkit/networkit/blob/master/networkit/cpp/auxiliary/Random.cpp#L26).

If (i) does not hold, then `getSeed()` behaves exactly as you did here. If (i) holds and (ii) does not, then each thread will be assigned a different seed. So my recommendation is to use `Aux::Random::integer()` only when both (i) and (ii) hold. Doing so, we avoid assigning the same seed to every thread, and the user can choose the base seed if they want.

Also, `baseSeed` can be `const`.
```cpp
const auto baseSeed = (GlobalState::getGlobalSeed() != 0 && !GlobalState::getSeedUseThreadId()) ?
  Aux::Random::integer() : Aux::Random::getSeed();
```

Reply:
@Relux-the-Relux do you agree on this? I think that once this is done we can merge the PR.

Reply:
@Relux-the-Relux @angriman Hopefully that is ok - I agree with the suggested change and updated the PR accordingly. If the PR is ok now, we can integrate it in the release then.